genesyscloud/routing_utilization_label/genesyscloud_routing_utilization_label_proxy.go

```diff
@@ -0,0 +1,139 @@
+
+func (p *routingUtilizationProxy) getRoutingUtilizationLabelByName(ctx context.Context, name string) (*platformclientv2.Utilizationlabel, *platformclientv2.APIResponse, error) {
+	return p.getRoutingUtilizationLabelByNameAttr(ctx, p, name)
+}
+
+func (p *routingUtilizationProxy) updateRoutingUtilizationLabel(ctx context.Context, id string, req *platformclientv2.Updateutilizationlabelrequest) (*platformclientv2.Utilizationlabel, *platformclientv2.APIResponse, error) {
+	return p.updateRoutingUtilizationLabelAttr(ctx, p, id, req)
+}
+
+func (p *routingUtilizationProxy) deleteRoutingUtilizationLabel(ctx context.Context, id string, forceDelete bool) (*platformclientv2.APIResponse, error) {
+	return p.deleteRoutingUtilizationLabelAttr(ctx, p, id, forceDelete)
+}
+
+func getAllRoutingUtilizationLabelsFn(_ context.Context, p *routingUtilizationProxy, name string) (*[]platformclientv2.Utilizationlabel, *platformclientv2.APIResponse, error) {
+	var allUtilizationLabels []platformclientv2.Utilizationlabel
+	const pageSize = 100
+
+	labels, resp, err := p.routingApi.GetRoutingUtilizationLabels(100, 1, "", name)
+	if err != nil {
+		return nil, resp, fmt.Errorf("failed to get routing utilization labels | error: %s", err)
+	}
+
+	if labels.Entities == nil || len(*labels.Entities) == 0 {
+		return &allUtilizationLabels, resp, nil
+	}
+	allUtilizationLabels = append(allUtilizationLabels, *labels.Entities...)
+
+	for pageNum := 2; pageNum <= *labels.PageCount; pageNum++ {
+		labels, resp, err := p.routingApi.GetRoutingUtilizationLabels(pageSize, pageNum, "", name)
+		if err != nil {
+			return nil, resp, fmt.Errorf("failed to get routing utilization labels | error: %s", err)
+		}
+
+		if labels.Entities == nil || len(*labels.Entities) == 0 {
+			break
+		}
+		allUtilizationLabels = append(allUtilizationLabels, *labels.Entities...)
+	}
+
+	for _, label := range allUtilizationLabels {
+		rc.SetCache(p.routingCache, *label.Id, label)
+	}
+
+	return &allUtilizationLabels, resp, nil
+}
+
+func createRoutingUtilizationLabelFn(_ context.Context, p *routingUtilizationProxy, req *platformclientv2.Createutilizationlabelrequest) (*platformclientv2.Utilizationlabel, *platformclientv2.APIResponse, error) {
+	return p.routingApi.PostRoutingUtilizationLabels(*req)
+}
+
+func getRoutingUtilizationLabelFn(_ context.Context, p *routingUtilizationProxy, id string) (*platformclientv2.Utilizationlabel, *platformclientv2.APIResponse, error) {
+	if label := rc.GetCacheItem(p.routingCache, id); label != nil {
+		return label, nil, nil
+	}
+
+	return p.routingApi.GetRoutingUtilizationLabel(id)
+}
+
+func getRoutingUtilizationLabelByNameFn(ctx context.Context, p *routingUtilizationProxy, name string) (*platformclientv2.Utilizationlabel, *platformclientv2.APIResponse, error) {
+	labels, resp, err := getAllRoutingUtilizationLabelsFn(ctx, p, name)
+	if err != nil {
+		return nil, resp, fmt.Errorf("error retrieving routing utilization label by name %s", err)
+	}
+
+	label := (*labels)[0]
```

Comment:
More that one label might be returned by the api. Look through the array of labels returned and do an exact check on the name before returning